keyboards/tetris/keymaps/default/keymap.c

```diff
@@ -0,0 +1,747 @@
+bool  comm_shifted  = false;
+bool  ques_shifted  = false;
+static uint8_t  key_index  = 0;
+
+bool process_record_user( uint16_t keycode, keyrecord_t *record )
+{
+  uint8_t    shifted;
+  uint16_t  s_keycode;
+  bool    *k_shifted;
+
+//单点亮
+if ( RGB_TAP_STATE ==true ) {
+  key_index=(record->event.key.col)+(record->event.key.row)*12;
+
+if (timer_elapsed(TAP_RGB_timer) >10000){
+TAP_RGB_timer=timer_read();
+Type_Hue=270;
+}else if (timer_elapsed(TAP_RGB_timer) >1000){
+Type_Hue+=30;
+if (Type_Hue>270) {
+Type_Hue=270;}
+}else{
+TAP_RGB_timer=timer_read();
+Type_Hue-=10;
+if (Type_Hue<10) {
+Type_Hue=10;}
+}
+
+  if (key_index>42)
+    {
+      key_index=key_index-1;
+    }
+  if ( record->event.pressed  )
+    {
+      //rgblight_setrgb_at(RGB_TAP_KEY_Color,key_index);
+      rgblight_sethsv_at(Type_Hue,255,255,key_index);
+    } else 
+    {
+      rgblight_setrgb_at(RGB_TAP_Base_Color,key_index);
+
+    }
+}
+
+  switch ( keycode )
+  {
+/* 记录rgb 设置 */
+  case RGB_M_R:
+    save_rgbmode = true;
+    break;
+  case RGB_M_SW:
+    save_rgbmode = true;
+    break;
+  case RGB_M_X:
+    save_rgbmode = true;
+    break;
+  case RGB_M_G:
+    save_rgbmode = true;
+    break;
+
+  case RGB_MODE_KNIGHT:
+    save_rgbmode = true;
+    break;
+  case KC_00:
+  if (record->event.pressed) {
+        // Do something when pressed
+      } else {
+      register_code( KC_0 );
+      unregister_code( KC_0 );
+      register_code( KC_0 );
+      unregister_code( KC_0 );
+      }
+      return false; // Skip all further processing of this key
+
+case RESET:
+rgblight_mode( 1 );
+    rgblight_setrgb( 0, 0, 0 );
+    rgblight_setrgb_at(128, 0, 0,4);
+    rgblight_setrgb_at(6, 128, 24,36);
+    break;
+case RGB_TAP:
+if ( !(record->event.pressed))
+{
+if (RGB_TAP_STATE) 
+{
+RGB_TAP_STATE=false;
+rgblight_mode( 1 );
+rgblight_setrgb( RGB_TAP_Base_Color );
+rgblight_setrgb_at(RGB_TAP_Off_Color,16);
+}
+else
+{
+RGB_TAP_STATE=true;
+rgblight_mode( 1 );
+rgblight_setrgb( RGB_TAP_Base_Color );
+rgblight_setrgb_at(RGB_TAP_On_Color,9);
+}
+}
+break;
+
+  /* 特殊 shift */
+  case KC_ENT:
+    s_keycode  = KC_QUOT;
+    k_shifted  = &ques_shifted;
+    break;
+  case KC_UP:
+    s_keycode  = KC_SLSH;
+    k_shifted  = &ques_shifted;
+    break;
+  case KC_DOWN:
+    s_keycode  = KC_BSLS;
+    k_shifted  = &ques_shifted;
+    break;
+  default:
+    return(true);
+  }
+
+  shifted = get_mods() & (MOD_BIT( KC_LSHIFT ) | MOD_BIT( KC_RSHIFT ) );
+
+  /* Keydown. If shift is currently pressed, register its alternate keycode. */
+  if ( record->event.pressed && shifted )
+  {
+    *k_shifted = true;
+    register_code( s_keycode );
+    return(false);
+    /*
+     * Keyup. If shift was pressed back when the key was pressed, unregister
+     * its alternate keycode.
+     */
+  } else if ( !(record->event.pressed) && *k_shifted )
+  {
+    *k_shifted = false;
+    unregister_code( s_keycode );
```

Comment:
Travis CI is erroring out on this.  "may be used uninitialized in this function"